src/crimson/os/seastore/lba/btree_lba_manager.cc

```diff
@@ -316,64 +316,42 @@ BtreeLBAManager::alloc_extents(
   DEBUGT("{}", t, pos);
   assert(pos.is_viewable());
   auto c = get_context(t);
-  return with_btree<LBABtree>(
-    cache,
-    c,
-    [c, FNAME, pos=std::move(pos), this,
-    extents=std::move(extents)](auto &btree) mutable {
-    auto &cursor = pos.get_effective_cursor();
-    return cursor.refresh(
-    ).si_then(
-      [&cursor, &btree, extents=std::move(extents),
-      pos=std::move(pos), c, FNAME, this] {
-      return seastar::do_with(
-	std::move(extents),
-	btree.make_partial_iter(c, cursor),
-	std::vector<LBAMapping>(),
-	[c, &btree, FNAME, this]
-	(auto &extents, auto &iter, auto &ret) mutable {
-	return trans_intr::do_for_each(
-	  extents.rbegin(),
-	  extents.rend(),
-	  [&btree, FNAME, &iter, c, &ret, this](auto ext) {
-	  assert(ext->has_laddr());
-	  stats.num_alloc_extents += ext->get_length();
-	  return btree.insert(
-	    c,
-	    iter,
-	    ext->get_laddr(),
-	    lba_map_val_t{
-	      ext->get_length(),
-	      ext->get_paddr(),
-	      EXTENT_DEFAULT_REF_COUNT,
-	      ext->get_last_committed_crc()}
-	  ).si_then([ext, c, FNAME, &iter, &ret](auto p) {
-	    auto &[it, inserted] = p;
-	    ceph_assert(inserted);
-	    auto &leaf_node = *it.get_leaf_node();
-	    leaf_node.insert_child_ptr(
-	      it.get_leaf_pos(),
-	      ext.get(),
-	      leaf_node.get_size() - 1 /*the size before the insert*/);
-	    TRACET("inserted {}", c.trans, *ext);
-	    ret.emplace(ret.begin(), LBAMapping::create_direct(it.get_cursor(c)));
-	    iter = it;
-	  });
+  auto btree = co_await get_btree(t);
+  co_await pos.get_effective_cursor().refresh();
+  auto iter = btree.make_partial_iter(c, pos.get_effective_cursor());
+  std::vector<LBAMapping> ret;
+  for (auto eit = extents.rbegin(); eit != extents.rend(); eit++) {
```

Comment:
using for loop instead of trans_intr::do_for_each will not handle interruptions.